modules/social_features/social_group/modules/social_group_request/social_group_request.install

```diff
@@ -21,9 +21,14 @@ function social_group_request_install() {
  * Assign group request permission for closed and flexible group types.
  */
 function _social_group_request_set_permissions() {
-  $group_type = GroupType::load('flexible_group');
-  $group_type->getOutsiderRole()->grantPermissions(['request group membership'])->save();
 
-  $group_type = GroupType::load('closed_group');
-  $group_type->getOutsiderRole()->grantPermissions(['request group membership'])->save();
+  if (\Drupal::moduleHandler()->moduleExists('social_group_flexible_group')) {
+    $group_type = GroupType::load('flexible_group');
+    $group_type->getOutsiderRole()->grantPermissions(['request group membership'])->save();
+  }
+
+  if (\Drupal::moduleHandler()->moduleExists('social_group_secret')) {
```

Comment:
@diakivvasyl This condition is not needed, since `closed_group` type is provided by `social_group` module and is enabled by default.